### PR TITLE
Fix sign up component's browser error bubble issue

### LINKED ISSUE #3051: Sign-up component bug - email validation error for invalid email should be inline
**Describe the bug**
According to design - email validation error should be inline and not appear in a browser default error bubble.

**To Reproduce**
1. Go to https://foundation.mozilla.org
2. Scroll down to the footer section
3. Type something random in email field

**Expected behavior**
Error message for invalid email should be inline.

**Screenshots**
sign up error bubble

### PR DESCRIPTION
**Pull request by mmmavis**
Closes #3051

### CODE REVIEW

File: source/js/components/join/join.jsx [hunk 1]
```diff
@@ -48,6 +48,18 @@ export default class JoinUs extends React.Component {
     );
   }
 
+  /**
```
Comment (mmmavis) [inline source/js/components/join/join.jsx]: Copied from `petition.jsx`: https://github.com/mozilla/foundation.mozilla.org/blob/master/source/js/components/petition/petition.jsx#L262-L272

**Comment by mmmavis** [comment]
@alanmoo the bug this PR is trying to fix is to prevent the browser bubble from showing up. If you check footer sign up on prod the submit button is already doing the things you described in your comment above. If we wanna fix it I think it should be tackled as a separate ticket.

**Comment by mmmavis** [comment]
Sorta related: I filed a ticket a while back for aligning sign up / petition form on Foundation site: https://github.com/mozilla/foundation.mozilla.org/issues/3005

**Comment by mmmavis** [comment]
@alanmoo nvm i misunderstood your comment lol.

File: source/js/components/join/join.jsx (hunk 1 above)
Comment (gideonthomas) [reply source/js/components/join/join.jsx]: interesting...wonder why we didn't use dummy form field validation i.e. something like:

```js
function isValidEmail(email) {
  const el = document.createElement('input');
  el.type = `email`;
  el.required = true;
  el.value = email;
  return el.validity.valid;
}
```

either way, this is still fine for majority emails.

Comment (mmmavis) [reply source/js/components/join/join.jsx]: I'll mention this in this ticket: #3005

**Pull request merged**